### PR TITLE
KAFKA-9025: ZkSecurityMigrator not working with zookeeper chroot

### PR DESCRIPTION
**Pull request by huxihx**
https://issues.apache.org/jira/browse/KAFKA-9025

If a chroot is configured, ZkSecurityMigrator should prompt a confirm to user to ensure whether chroot is specified correctly.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by huxihx** [comment]
@guozhangwang Please review this patch. Thanks.

**Comment by huxihx** [comment]
retest this please

**Comment by huxihx** [comment]
retest this please

**Comment by huxihx** [comment]
@hachikuji Please review this patch. Thanks.

**Review by guozhangwang — COMMENTED** [review]
Not sure I've fully captured ZkSecurityMigrator, @omkreddy could you take a look?

File: core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala [hunk 1]
```diff
@@ -61,7 +62,7 @@ object ZkSecurityMigrator extends Logging {
                       + "znodes as part of the process of setting up ZooKeeper "
                       + "authentication.")
 
-  def run(args: Array[String]): Unit = {
+  def run(args: Array[String], checkPathExists: Boolean = true): Unit = {
```
Comment (omkreddy) [inline core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala]: Normally users use `./bin/zookeeper-security-migration.sh` script for migration. Adding method parameter may not be useful. maybe we can enable path check by using cmd line option.

**Comment by huxihx** [comment]
retest this please.

**Comment by huxihx** [comment]
retest this please.

**Comment by guozhangwang** [comment]
@huxihx Could you rebase, so that @omkreddy could take another look and possibly merge?

**Review by omkreddy — APPROVED** [review]
@huxihx Thanks for the PR. LGTM.

**Pull request closed without merging**